handlers/healthcheck_finance.go

```diff
@@ -27,53 +27,87 @@ func HandleHealthCheckFinanceSystem(w http.ResponseWriter, r *http.Request) {
 	var systemAvailableTime time.Time
 
 	// Check for weekly downtime
-	if cfg.WeeklyMaintenanceStartTime != "" && cfg.WeeklyMaintenanceEndTime != "" {
```

Comment:
Attempt to reduce complexity by grouping multiple logic checks into one Go function